File: java/medium/Division.java
package java.medium;

public class Division {
    public static int MathChallenge(int num1, int num2) {
        int min=num1<num2?num1:num2;
        int result=1;
        for(int i=1;i<=min;i++){
          if(num1%i==0 && num2%i==0){
            result=i;
          }
        }
        return result;
      }
}

File: javascript/medium/division.js
let num1=108
let num2=24
let min=num1<num2?num1:num2
let result=1
for (let i = 1; i <=min; i++) {
    if (num1%i==0 && num2%i==0) {
        result=i
    }
}
console.log(result)